### PR TITLE
Add support for Ubuntu 16.04 and systemd

### PR DESCRIPTION
**Pull request by MiLk**
I've added the systemd service file to make Vault work with systemd on Ubuntu 16.04.

I've also replaced the command module by unarchive to unzip the zipfile with the vault binary.

### CODE REVIEW

**Comment by MiLk** [comment]
@kbrebanov I've updated my PR to use your latest changes.
But I think all the `become: yes` should be removed and moved to the playbook level.

Do you want me to do other changes before merging this PR?

File: tasks/main.yml [hunk 1]
```diff
@@ -103,6 +105,26 @@
   tags:
     - vault
 
+- name: Detect if systemd is present
```
Comment (kbrebanov) [inline tasks/main.yml]: This task can be removed and check if the distribution release is xenial instead.

File: tasks/main.yml [hunk 2]
```diff
@@ -103,6 +105,26 @@
...
+
+- name: Create Vault systemd configuration
+  become: yes
+  template:
+    src: vault.service.j2
+    dest: /etc/systemd/system/vault.service
+    owner: root
+    group: root
+    mode: 0644
+  when: vault_systemctl_stat.stat.exists is defined and vault_systemctl_stat.stat.exists == True
```
Comment (kbrebanov) [inline tasks/main.yml]: Just check if it's xenial:

ansible_distribution_release|lower == 'xenial'

File: tasks/main.yml [hunk 3]
```diff
@@ -111,6 +133,7 @@
     owner: root
     group: root
     mode: 0644
+  when: vault_systemctl_stat.stat.exists is not defined or vault_systemctl_stat.stat.exists == False
```
Comment (kbrebanov) [inline tasks/main.yml]: check if it's Trusty:

ansible_distribution_release|lower == 'trusty'

File: templates/vault.service.j2 [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Vault server
+After=network.target
+
+[Service]
+ExecStart={{ vault_install_dir }}/vault server -config {{ vault_config_dir }}/vault.hcl
+StandardOutput=journal
```
Comment (kbrebanov) [inline templates/vault.service.j2:7]: Would it be possible to send the logs to /var/log/vault.log like the Upstart script does? To have some consistency.

**Review by kbrebanov — CHANGES_REQUESTED** [review]
It would be great if this would be split into 2 different commits.

One for the unarchive module and one for adding support for Xenial.

You should also uncomment the environment section for Xenial in the Travis configuration file so that the tests will be executed on a Xenial instance.

As for the become directives. I disagree with simply using it at the playbook level. To me, that is the same bad practice as logging in as root when you don't need to. Only tasks that need elevated privileges should do so.

**Comment by MiLk** [comment]
I didn't use `ansible_distribution_release` to avoid relying on fact, but you already use `ansible_distribution`,  so it's ok.

I've configured the systemd service file to send the logs to a file instead of the journal, but it's still possible to override the `vault_log_file` with `journal` to send the logs to `journald` which is consistent across all the services running on a system using systemd.

**Comment by kbrebanov** [comment]
Could you just use a default variable named 'use_systemd_journal' and set it to 'false' by default?
Then all you would need is a conditional in the template.

You could probably also get rid of the following tasks:
- Check if the log file is present
- Create vault log file

Don't forget to also update **defaults/main.yml** and **README.md** accordingly.

**Comment by MiLk** [comment]
The ~~consul~~ _vault_ user can not write a new file into the `/var/log` folder (owner by `root.syslog` with permissions `775`), that's why I had to add these two tasks. If not the service does not start.

**Comment by kbrebanov** [comment]
I'm sorry. I don't understand what the **consul** user has to do with this role.
Are you applying this role on your machine using the **consul** user?

This role only has a dependency on **unzip**. Consul is not required to apply this role.
Also, the Travis CI tests will execute as root so it should have no issues creating the log file.

**Comment by MiLk** [comment]
Right, `vault` user, not `consul`.

The [command executed](https://github.com/MiLk/ansible-vault/blob/systemd/templates/vault.service.j2#L10) by systemd will be run with the [`vault` user](https://github.com/MiLk/ansible-vault/blob/systemd/templates/vault.service.j2#L14-L15).
The [`vault_log_file`](https://github.com/MiLk/ansible-vault/blob/systemd/vars/main.yml#L9) being in `/var/log` and the `vault` user not being able to write in this directory, the command fails. 

```
Oct 27 03:58:50 consul003 systemd[1]: Started Vault server.
Oct 27 03:58:50 consul003 sh[18224]: /bin/sh: 1: cannot create /var/log/vault.log: Permission denied
Oct 27 03:58:50 consul003 systemd[1]: vault.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Oct 27 03:58:50 consul003 systemd[1]: vault.service: Control process exited, code=exited status=1
Oct 27 03:58:50 consul003 systemd[1]: vault.service: Unit entered failed state.
Oct 27 03:58:50 consul003 systemd[1]: vault.service: Failed with result 'exit-code'.
Oct 27 03:58:50 consul003 systemd[1]: vault.service: Service hold-off time over, scheduling restart.
```

**Comment by kbrebanov** [comment]
Ok. Since it's best practice to log to the journal with systemd, let's just log to it.
You can remove everything that relates to logging somewhere else than the journal and the task that creates the log file.

**Comment by MiLk** [comment]
I've removed the 2 commits related to the log file.

**Review by kbrebanov — APPROVED** [review]

**Pull request merged**